Parser/Node.cs

```diff
@@ -28,6 +28,17 @@ public record NDeclarations (NVarDecl[] Vars) : Node {
 public record NVarDecl (Token Name, NType Type) : Node {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
+
+// Declares a variable (with a type)
+public record NProcDecl (Token Name, NVarDecl[] Vars, NBlock Block) : Node {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// Declares a variable (with a type)
+public record NFuncDecl (Token Name, NVarDecl[] Vars, NType Type, NBlock Block) : Node {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+

```

Comment:
Try to combine both into a single NFuncDecl record. If it's a procedure, we'll denote that by setting the return type to NType.Void. Most of the parsing / calling / code generation for functions and procedures will be similar so it makes sense to have a single record to represent both.